docs/superpowers/plans/2026-06-08-opentelemetry-instrumentation.md

```
@@ -0,0 +1,2241 @@
+	// After the round trip:
+	spans := rec.Ended()
+	if len(spans) < 2 {
+		t.Fatalf("expected >=2 spans (client+server), got %d", len(spans))
+	}
+	// Find the server span and assert its parent TraceID == client span TraceID.
+	// (Concrete assertion completed once the ServerService.Health handler is wired.)
+}
+```
+
+> **Implementer note:** complete the handler/client wiring using `gen/specgraph/v1/specgraphv1connect` — the `ServerService.Health` RPC (`/specgraph.v1.ServerService/Health`, request `*v1.HealthRequest`) is the simplest, requiring no project header. Embed `specgraphv1connect.UnimplementedServerServiceHandler` in your stub and override only `Health`. The assertion to finish: locate the server-kind span and the client-kind span via `span.SpanKind()`, then assert `serverSpan.Parent().SpanID() == clientSpan.SpanContext().SpanID()` and equal `TraceID()`. The `otelconnect.WithTracerProvider`/`WithPropagator` option names must be verified against v0.9.0.
+
```

markdownlint-cli2 (0.22.1)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)